### DISCUSSION TITLE
Bookmarks lost at filename change

### DISCUSSION THREAD

**Discussion by Wilmvanrooij** [alefragnani/vscode-bookmarks#593] [Q&A] [answered]
Hi all,

At times, I find myself needing to modify a file's name in vsc, and when I do so, I lose the associated bookmarks. While I'm aware that I can manually update the bookmarks.json file, I'm wondering if there is a more convenient way to transfer the bookmarks from the original file to the renamed version ?

Thank you for your help and any suggestions on this topic would be greatly appreciated.

Wilm

**Comment by alefragnani** [comment, marked as answer]
Hi @Wilmvanrooij ,

If you rename the file inside VS Code, using the native `Rename` context menu item, the bookmarks will be automatically updated as well. This is possible because VS Code _notifies_ file renaming and I can handle it. This feature has been originally requested by #209, and released a few years ago.

Renaming the file outside VS Code, shell or other tools/commands, isn't supported.

Hope this helps
